### PR TITLE
TDF/TSF Support

### PR DESCRIPTION
**Pull request by SteffenHeu**
https://github.com/mzmine/mzmine3/projects/2

### CODE REVIEW

**Comment by SteffenHeu** [comment]
The FrameToFile module is for testing purposes and has to be removed before merging.

**Comment by tomas-pluskal** [comment]
I merged Roman's PR, which created a few merge conflicts here, please take a look.

File: src/main/java/io/github/mzmine/datamodel/IMSRawDataFile.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
+package io.github.mzmine.datamodel;
```
Comment (tomas-pluskal) [inline src/main/java/io/github/mzmine/datamodel/IMSRawDataFile.java:1]: Please add the license headers to all new java files :)

File: src/main/java/io/github/mzmine/modules/io/projectsave/PeakListElementName.java [hunk 2]
```diff
@@ -20,24 +20,44 @@
...
-                                      "applied_method"), METHOD_NAME(
-                                          "method_name"), METHOD_PARAMETERS(
-                                              "method_parameters"), REPRESENTATIVE_SCAN(
-                                                  "best_scan"), FRAGMENT_SCAN(
-                                                      "fragment_scan"), ALL_MS2_FRAGMENT_SCANS(
-                                                          "all_MS2_fragment_scans"), INDEX(
-                                                              "index"), SHARPNESS(
-                                                                  "sharpness"), PARENT_CHROMATOGRAM_ROW_ID(
-                                                                      "parent_chromatogram_row_id");
+  PEAKLIST("peaklist"),
```
Comment (tomas-pluskal) [inline src/main/java/io/github/mzmine/modules/io/projectsave/PeakListElementName.java:23]: The autoformatter in Eclipse or IDEA is going to kill the formatitng of these enums. I recommend to use this syntax:
````
  PEAKLIST("peaklist"), //
  PEAKLIST_DATE("created"), //
  QUANTITY("quantity"), //
````
That way the newlines will be preserved. The same applies to your other enums (BrukerScanMode etc.).

Comment (SteffenHeu) [reply src/main/java/io/github/mzmine/modules/io/projectsave/PeakListElementName.java:23]: The formatter didnt change it for me, but you're right, better make it safe for everyone

Comment (SteffenHeu) [reply src/main/java/io/github/mzmine/modules/io/projectsave/PeakListElementName.java]: Done. I hope i found all of them

**Comment by tomas-pluskal** [comment]
Thanks! Should I merge this now, so we have all the major changes in the data model included?

File: src/main/java/io/github/mzmine/datamodel/Frame.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+package io.github.mzmine.datamodel;
+
+import com.google.common.collect.Range;
+import java.util.List;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+/**
+ * A frame is a collection of mobility resolved spectra at one point in time.
```
Comment (tomas-pluskal) [inline src/main/java/io/github/mzmine/datamodel/Frame.java:27]: Just curious, does "at one point in time" mean in retention time? Cause "time" can have different meanings.

Comment (SteffenHeu) [reply src/main/java/io/github/mzmine/datamodel/Frame.java:27]: yes, RT

**Comment by SteffenHeu** [comment]
> 
> 
> Thanks! Should I merge this now, so we have all the major changes in the data model included?

Yeah, you can merge it.

**Pull request merged**